File: src/main/java/com/damon140/ur/Team.java
package com.damon140.ur;

// TODO: light & dark as less racist names??
public enum Team {
    white, black;

    public final String ch;

    Team() {
        ch = this.name().substring(0, 1);
    }

    public Team other() {
        return Team.values()[(this.ordinal() + 1) % 2];
    }

    public static boolean isTeamChar(String value) {
        return value.equals(white.ch) || value.equals(black.ch);
    }

    public static Team fromCh(String value) {
        return white.ch.equals(value) ? white : black;
    }
}

File: hello2/src/main/kotlin/ur/Team.kt
package com.damon140.ur

enum class Team {
    white, black;

    val ch: String

    init {
        ch = name.substring(0, 1)
    }

    fun other(): Team {
        return values()[(ordinal + 1) % 2]
    }

    companion object {
        fun isTeamChar(value: String): Boolean {
            return value == white.ch || value == black.ch
        }

        fun fromCh(value: String): Team {
            return if (white.ch == value) white else black
        }
    }
}